Utilities/DavixAdaptor/src/DavixFile.cc

```diff
@@ -0,0 +1,363 @@
+}
+
+void DavixFile::create(const std::string &name, bool exclusive /* = false */,
+                       int perms /* = 066 */) {
+  open(name.c_str(), (IOFlags::OpenCreate | IOFlags::OpenWrite | IOFlags::OpenTruncate |
+                      (exclusive ? IOFlags::OpenExclusive : 0)),
+       perms);
+}
+
+void DavixFile::open(const std::string &name, int flags /* = IOFlags::OpenRead */,
+                     int perms /* = 066 */) {
+  open(name.c_str(), flags, perms);
+}
+
+void DavixFile::open(const char *name, int flags /* = IOFlags::OpenRead */, int perms /* = 066 */) {
+  // Actual open
+  if ((name == 0) || (*name == 0)) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Cannot open a file without name";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+
+  if ((flags & IOFlags::OpenRead) == 0) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Must open file '" << name << "' at least for read";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+
+  if (davixPosix && m_fd) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Davix::open(name='" << m_name->c_str() << "') failed on already open file";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+  configureDavixLogLevel();
+  // Translate our flags to system flags
+  int openflags = 0;
+
+  if (flags & IOFlags::OpenRead)
+    openflags |= O_RDONLY;
+
+  DavixError *davixErr = NULL;
+  davixReqParams = std::make_unique<RequestParams>();
+  // Set up X509 authentication
+  davixReqParams->setClientCertCallbackX509(&X509Authentication, NULL);
+  // Set also CERT_DIR if it is set in envinroment, otherwise use default
+  const char *cert_dir = NULL;
+  if ((cert_dir = getenv("X509_CERT_DIR")) == NULL)
+    cert_dir = "/etc/grid-security/certificates";
+  davixReqParams->addCertificateAuthorityPath(cert_dir);
+
+  davixPosix = std::make_unique<DavPosix>(new Context());
+  m_fd = davixPosix->open(davixReqParams.get(), name, openflags, &davixErr);
+
+  m_name = std::make_unique<std::string>(name);
+  // Check Davix Error
+  if (davixErr) {
+    std::unique_ptr<DavixError> davixErrManaged(davixErr);
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Davix::open(name='" << m_name->c_str() << "') failed with "
+       << "error '" << davixErr->getErrMsg().c_str() << " and error code " << davixErr->getStatus();
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+  if (!m_fd) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Davix::open(name='" << m_name->c_str() << "') failed as fd is NULL";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+}
+
+IOSize DavixFile::readv(IOBuffer *into, IOSize buffers) {
+  assert(!buffers || into);
+
+  // Davix does not support 0 buffers;
+  if (buffers == 0)
+    return 0;
+
+  DavixError *davixErr = NULL;
+
+  DavIOVecInput input_vector[buffers];
+  DavIOVecOuput output_vector[buffers];
```

Comment:
@Dr15Jones I think, @juztas already was using Clang during the development to iron out possible issues and it did compile. Clang does support VLA (.. or to some undocumented scope). I think, Davix examples also contains VLAs.

We could replace them with allocation on heap instead and VLAs are gone.